### DISCUSSION TITLE
Any ETA on Ruby 2.7 support?

### DISCUSSION THREAD

**Discussion by JedMeister** [TracksApp/tracks#2827] [Q&A] [answered]
Hi there,

As I understand it, currently Tracks requires [Ruby 2.5](https://github.com/TracksApp/tracks/blob/master/.ruby-version). But I note that Ruby 2.5 is getting pretty old now.

It appears that there was an attempt to update earlier in the year, but it was rolled back because of [rubocop](https://github.com/TracksApp/tracks/commit/f99b423445580a611913f372114d74cc9329695a).

I'm no Ruby expert by any stretch, but my reading of the [RuboCop docs](https://docs.rubocop.org/rubocop/compatibility.html) suggests that it supports "MRI 2.6+". From what I understand, "MRI" is the reference Ruby implementation, suggesting it should work with v2.7 (I know it doesn't always work like that, but...).

I also note that [RuboCop](https://rubocop.org/) appears to be a static code analysis tool. So whilst having it as part of the build/dev chain makes tons of sense, does it even need to be part of the installed project? Perhaps I completely misunderstand, but not including build time dependencies (that aren't also runtime dependencies) seems like a reasonable approach to me.

Bottom line though, if it's not going to change anytime soon I'll push ahead using Ruby 2.5. But if it's going to be updated soonish, then I might wait.

And whilst I've got close to zero experience with Ruby, if there is something I can do to help, please ask.

**Comment by ZeiP** [comment, marked as answer]
After the last release the only supported Ruby versions are indeed 2.7 and 3.0. 3.1 is not yet supported but will hopefully be before the next release. So you can (and should) absolutely use 2.7 or 3.0. The default Docker image uses 2.7. What kind of problems did you have in using 2.7?

The question concerning removing RuboCop from the release is a good one; I haven't really taken a good look at the dependencies but I should.

**Reply by ZeiP** [reply]
Ah yes, now I see what you meant with the question! The .rubyver file is meant to only limit the Ruby version used by Codeclimate, as that uses a Rubocop only supporting versions up to 2.5: https://codeclimate.com/github/TracksApp/tracks/pull/2828  It shouldn't, however, limit the Ruby version actually used to run the software.

**Reply by JedMeister** [reply]
Thanks! It sounds like I completely misunderstood!

A colleague (who is much more experienced with Ruby than me) noted that earlier in the year he couldn't get it to work with any Ruby version other than 2.5. My guess is that either it was specific to that point in time (and Tracks has since been improved to work with newer Ruby) or something else (unrelated specifically to Tracks) was wrong in our environment (and my colleague jumped to the wrong conclusion).

Anyway, thanks for clearing that up. Legend! :sunglasses:

**Reply by ZeiP** [reply]
Tracks has supported Ruby 2.7 at least at the beginning of 2021: https://github.com/TracksApp/tracks/pull/2545/checks (which was tested with ruby 2.5, 2.6 and 2.7) So yeah, it might've been something else. Let me know if you have any problems with Tracks!

**Reply by JedMeister** [reply]
Will do, thanks! :smile:
